airbyte-webapp/src/services/connector/useSuspenseQuery.ts

```diff
@@ -0,0 +1,21 @@
+import { QueryFunction, QueryKey, useQuery, UseQueryOptions } from "react-query";
+
+type Disabled = Readonly<{ enabled: false }>;
+
+export function useSuspenseQuery<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey
+>(
+  queryKey: TQueryKey,
+  queryFn: QueryFunction<TQueryFnData, TQueryKey>,
+  options: Readonly<
+    Omit<UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>, "queryKey" | "queryFn" | "suspense">
+  > = {}
+) {
+  return useQuery<TQueryFnData, TError, TData, TQueryKey>(queryKey, queryFn, {
+    ...options,
+    suspense: true,
+  }).data as typeof options extends Disabled ? TData | undefined : TData;
```

Comment:
We sometimes disable the call due to the rule of hooks. This lets typescript correctly infer the return type.